test/dynamo/test_export.py

```diff
@@ -2275,6 +2275,30 @@ def forward(self, *args):
         res = gm(input_tensor, input_tensor2)
         self.assertTrue(torch._dynamo.utils.same(ref, res))
 
+    @config.patch(dynamic_shapes=True)
```

Comment:
I don't see why you need dynamic shapes = True; the whole point of static nonzero is that it works without dynamic shapes.

Reply:
Yeah, need clarification here. 

IIUC we will (1) set `dynamic_shapes=True` by default for export eventually (2) we want to make the program using `nonzero_static()` exportable. That's why it's set True here.

Secondly the program with `nonzero_static()` is exportable (no error) when `dynamic_shapes=False` **w/o adding the meta function for it**, so if it's expected to be used with `dynamic_shapes=False` why bother adding a meta function then?

May not direct relevant to this comment, but I want to ask if we would like to make the `size` param `int` or `SymInt`? Currently I assume we would always want to burn the `size` statically to the exported program so making it a `int`, but I'm not sure whether we have use cases would like to make it `SymInt`

Reply:
> I don't believe you lol. How did you repro this.

@ezyang It's reproducible for both c++ impl and py impl. Here is a patch (D44514255) for c++ impl (ignore the dtoh sync problem). As you can see I explicitly set dynamic_shapes=False for export and commented out the meta function in _meta_registrations.py, the test passes.